### PR TITLE
Enable use for plain ruby classes

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Allow creation of GDS design system form components for plain ruby classes. [Issue](https://github.com/DFE-Digital/govuk_design_system_formbuilder/issues/88).

#### Why
Inline with standard rails form builder helpers and to facilitate creation of simple forms with no models associated.

For example to create a simple search form it would be nice to not have to create any model and just do something like 

```
# routes.rb
post 'search', action: :index, controller: :search
```

```
# app/controllers/search_controller.rb
class SearchController < ApplicationController
  def index
    @search_options = [[:opt1, 'Option 1'], [:opt2, 'Option 2']]
  end
end
```

```
# app/view/search/index.html.haml
= form_with url: search_path, local: true do |f|
  = f.govuk_collection_radio_buttons :search_option,
      @search_options,
      :first,
      :last,
      legend: { text: 'Choose a filter' },
      hint_text: 'Specify a filter for the search'
  = f.govuk_text_field(:query, label: { text: 'Find a thing' }, hint_text: 'enter thing to find' )
  = f.govuk_submit('Search')

```

Currently this is only prevented by the use of calls to `errors`, from what i can see.

### CODE REVIEW

File: lib/govuk_design_system_formbuilder/elements/error_summary.rb [hunk 1]
```diff
@@ -65,6 +65,8 @@ def error_summary_title_id
       end
 
       def object_has_errors?
+        return unless @builder.object.respond_to?(:errors)
```
Comment (peteryates) [inline lib/govuk_design_system_formbuilder/elements/error_summary.rb]: I think the builder _should_ error if the dev tries to render an error summary in a form with an object that doesn't support errors

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -57,9 +57,9 @@
   # triggering implicit auto-inclusion in groups with matching metadata.
   config.shared_context_metadata_behavior = :apply_to_host_groups
 
-# The settings below are suggested to provide a good initial experience
-# with RSpec, but feel free to customize to your heart's content.
-=begin
+  # The settings below are suggested to provide a good initial experience
```
Comment (peteryates) [inline spec/spec_helper.rb]: Happy for this to be amended but would prefer a separate dedicated PR

Comment (jsugarman) [reply spec/spec_helper.rb]: no worries. will push one

File: lib/govuk_design_system_formbuilder/elements/error_summary.rb (hunk 1 above)
Comment (jsugarman) [reply lib/govuk_design_system_formbuilder/elements/error_summary.rb]: > Overall this looks good. Before this was raised I'd already created #90 which does more or less the same thing. If you're happy for me to I'll cherry pick the tests from this PR into that one and tweak accordingly. Thanks @jsugarman

Yep, what ever you think is best.

Comment (jsugarman) [reply lib/govuk_design_system_formbuilder/elements/error_summary.rb]: > I think the builder _should_ error if the dev tries to render an error summary in a form with an object that doesn't support errors

👍 Makes sense, better to catch these things early.
